src/azul/plugins/metadata/anvil/service/response.py

```diff
@@ -181,44 +181,54 @@ def _non_pivotal_entity(self,
             if k in fields
         }
 
-    _non_pivotal_fields_by_entity_type = {
-        'activities': {
-            'activity_type',
-            'assay_type',
-            'data_modality'
-        },
-        'biosamples': {
-            'anatomical_site',
-            'biosample_type',
-            'disease',
-            'donor_age_at_collection_unit',
-            'donor_age_at_collection',
-        },
-        'datasets': {
-            'dataset_id',
-            'title'
-        },
-        'diagnoses': {
-            'disease',
-            'phenotype',
-            'phenopacket',
-            'onset_age_unit',
-            'diagnosis_age_unit',
-            # These fields are of high cardinality but only appear as inner
-            # entities for donors, so the aggregation size should still be low.
-            'diagnosis_age',
-            'onset_age',
-        },
-        'donors': {
-            'organism_type',
-            'phenotypic_sex',
-            'reported_ethnicity',
-            'genetic_ancestry'
-        },
-        'files': {
-            'count',
-            'data_modality',
-            'file_format',
-            'reference_assembly'
+    @property
```

Comment:
```suggestion
    @cached_property
```

since this is called quite frequently.

Reply:
https://github.com/DataBiosphere/azul/issues/4931

Reply:
> I would have liked to see the impact on the service response.

I realize it's not as informative/convenient as a git diff, but you can observe the update service response on [nadove4](https://service.nadove4.anvil.gi.ucsc.edu)